git.go

```diff
@@ -23,6 +24,14 @@ func init() {
 }
 
 func DecodeBlock(block blocks.Block) (node.Node, error) {
+	h, err := mh.Decode(block.Cid().Hash())
+	if err != nil {
+		return nil, err
+	}
+
+	if block.Cid().Type() != cid.GitRaw || h.Code != mh.SHA1 {
+		return nil, errors.New("invalid encoding")
+	}
```

Comment:
You'll also need to check the hash length (`-1` (because code) or `mh.DefaultLengths[mh.SHA1]`) and the CID version. I'd just call `block.Cid().Prefix()` and check everything in that struct.